primitives/core/src/traits.rs

```diff
@@ -188,51 +200,131 @@ impl TaskExecutorExt {
 
 /// Runtime spawn extension.
 pub trait RuntimeSpawn: Send {
+	/// Run a call as native.
+	///
+	/// Returns handle of the spawned task.
+	fn spawn_call_native(
+		&self,
+		func: fn(Vec<u8>) -> Vec<u8>,
+		data: Vec<u8>,
+		declaration: WorkerDeclaration,
+		calling_ext: &mut dyn Externalities,
+	) -> u64;
```

Comment:
Previously we did not have it because we weren't using a shared ref to spawn native (direct call when now we schedule it).